tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -102,6 +95,19 @@ public RecordEntry Lookup(RecordEntry requestEntry, RecordMatcher matcher, IEnum
             {
                 sanitizer.Sanitize(requestEntry);
             }
+            return LookUpSanitizedRecord(requestEntry, matcher, remove);
+        }
+
+        private RecordEntry LookUpSanitizedRecord(RecordEntry requestEntry, RecordMatcher matcher, bool remove)
+        {
+            try
+            {
+                // normalize request body with STJ using relaxed escaping to match behavior when Deserializing from session files
+                RecordEntry.NormalizeJsonBody(requestEntry.Request, requestEntry.Request.Body);
```

Comment:
We should do this at the place that builds the RecordEntry.

Reply:
this is the place where the sanitization occurs so it makes sense to keep here.